docs/analyzer/user_guide.md

```diff
@@ -343,6 +343,32 @@ Example:
 CodeChecker log -o ../codechecker_myProject_build.log -b "make -j2"
 ```
 
+### Change user inside the build command
+If we change user inside the build command of the CodeChecker log command
+before the actual compiler invocation, the compilation database will be empty:
+
+```sh
+CodeChecker log -o compile_commands.json -b "su myuser  -c 'g++ main.cpp -o /dev/null'"
+```
+
+The problem at here that the compilation database file and the lock file will
+be created with the user who runs the CodeChecker log command and only this
+user will have permission to read/write these files. So after changing to a
+different user, this user will not have permissions on these files.
+A solution can be if we change a user before the CodeChecker log command:
+
+```sh
+# Create a directory for compilation database.
+mkdir -p log_dir
+
+# Change file owner of log_dir.
+chown myuser log_dir
+
+# Run the command.
+su myuser  -c "CodeChecker log -o log_dir/compile_commands.json -b 'g++ main.cpp -o /dev/null'"
```

Comment:
I think we should use `su myuser -c ...` in the build command here, because the point of this documentation is that the user who runs the logger can't change the build command which runs it in the name of another user.

Reply:
As we discussed personally the problem here that the user would like to change user anyway.